Packs/Zerohack_XDR/Integrations/ZerohackXDR/ZerohackXDR.py

```diff
@@ -0,0 +1,331 @@
+                    }
+                    incidents.append(incident)
+                    last_incident_time = incident_created_time
+        if last_fetch == last_incident_time:
+            demisto.debug(f"Couldnt find new incidents with {last_fetch}. Updating.")
+            last_incident_time = last_incident_time + 1
+
+    next_run = {'last_fetch': last_incident_time}
+    return next_run, incidents
+
+def get_latest_incident(client: Client, severity_level: str):
+    """
+    This function is responsible for fetching a single sample incident for study/inspection purposes by the analyser or the SOAR handler.
+    It can be run in playground and it gives output in readable format so you can evaluate the incident format.
+
+    :param client: The client object to use for connection.
+    :type client: ``Client``
+
+    :param severity_level: This is the level of the severity you want to query the zerohack XDR for.
+    :type severity_level: ``int``
+
+    :return incident: List of incidents that will be created in XSOAR.
+    :rtype: ``List[dict]``
+    """
+
+    # Hard coding thge max results as we only need to send the latest output.
+    max_results = 1
+    alert = client.get_alert(severity_level=severity_level, max_results=max_results, offset=0)
+    incident_data = alert["data"][0]
+    incident_name = "Zerohack XDR " + incident_data['ids_threat_class']
+    incident = {
+        'name': incident_name,
+        'occurred': incident_data['attack_timestamp'],
+        'type': incident_data['ids_threat_class'],
+        'movement_type': incident_data['type_of_threat'],
+        'platform': incident_data['platform'],
+        'attacker_rep': incident_data['ip_rep'],
+        'rawJSON': json.dumps(str(incident_data)),
+        'severity': convert_to_demisto_severity(incident_data.get('ids_threat_severity', 'Low')),
+    }
+
+    return incident
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """
+    This function is the main control function.
+    It is responsible for handling the core control logic of the XDR integration.
+    This component handles the command input and fetching control. Apart from command control it alkso handles the inputs from the integration settings.
+    """
+
+    # Collecting details for initializing the connection.
+    api_key = demisto.params().get('apikey')
+    base_url = ZEROHACK_XDR_API_BASE_URL
+    verify_certificate = not demisto.params().get('insecure', False)
+    proxy = demisto.params().get('proxy', False)
+
+    # Determining the first fetch timestamp for default settings of integration.
+    first_fetch_time = arg_to_datetime(arg=demisto.params().get('first_fetch', '1 day'),
+                                       arg_name='First fetch time', required=True)
+    first_fetch_timestamp = int(first_fetch_time.timestamp()) if first_fetch_time else None
+
+    # First fetch timestamp failsafe.
+    assert isinstance(first_fetch_timestamp, int)
```

Comment:
Let's throw an indicative exception here-
try:
 int(first_fetch) 
except TypeError: 
 raise DemistoException("the first fetch value is invalid")